controllers/vaultauth_controller.go

```diff
@@ -64,13 +68,48 @@ func (r *VaultAuthReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	if o.GetDeletionTimestamp() != nil {
 		logger.Info("Got deletion timestamp", "obj", o)
+		r.referenceCache.Remove(VaultAuthGlobal, req.NamespacedName)
 		return r.handleFinalizer(ctx, o)
 	}
 
 	// assume that status is always invalid
 	o.Status.Valid = false
-
 	var errs error
+
+	var conditions []metav1.Condition
```

Comment:
Should this be adding onto the existing conditions on the object's status? It looks like this starts over with a single entry every evaluation.

Reply:
Should be resolved with https://github.com/hashicorp/vault-secrets-operator/pull/850